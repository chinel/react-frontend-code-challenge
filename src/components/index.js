export { default as DivisionSummary } from "./DivisionSummary/DivisionSummary";
export { default as Header } from "./Header/Header";
export { default as InfoBox } from "./InfoBox/InfoBox";
export { default as ModuleHistory } from "./ModuleHistory/ModuleHistory";
export { default as TableSummary } from "/TableSummary/TableSummary";
