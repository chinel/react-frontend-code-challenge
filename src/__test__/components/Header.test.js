import React from "react";
import { render } from "@testing-library/react";
import useStyles from "../../components/Header/HeaderStyles";
import { Header } from "../../components";
function checkStyles() {
  let values;

  function MockedComponent() {
    values = useStyles();
    return null;
  }
  render(<MockedComponent />);
  return values;
}

describe("<Header/>", () => {
  it("renders the <Header/> with populated data", () => {
    const { container, getByText } = render(<Header />);

    expect(getByText("FE Engineer Test")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it("validate the return values of useStyles hook", () => {
    const validateStlyes = checkStyles();

    expect(validateStlyes.menuBar).toBeTruthy();
  });
});
