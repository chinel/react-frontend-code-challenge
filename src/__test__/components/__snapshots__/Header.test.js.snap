// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders the <Header/> with populated data 1`] = `
<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-menuBar-1 mui-fixed MuiPaper-elevation4"
>
  <div
    class="MuiToolbar-root MuiToolbar-regular makeStyles-toolBar-2 MuiToolbar-gutters"
  >
    <div
      class="makeStyles-leftMenu-4"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="/"
      >
        <img
          alt="logo"
          src="/images/logo.png"
        />
        <p
          class="MuiTypography-root makeStyles-title-6 MuiTypography-body1"
        >
          FE Engineer Test
        </p>
      </a>
    </div>
    <div
      class="makeStyles-rightMenu-5"
    >
      <button
        aria-label="notifications"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-notificationButton-21 MuiIconButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <span
            class="MuiBadge-root makeStyles-notificationBadge-20"
          >
            <img
              alt="notifications"
              src="noti-Shape.svg"
            />
            <span
              class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary"
            >
              3
            </span>
          </span>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-haspopup="true"
        aria-label="user"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-buttonNav-24 MuiIconButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-userIcon-22"
            focusable="false"
            style="color: rgb(114, 128, 157);"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v1c0 .55.45 1 1 1h14c.55 0 1-.45 1-1v-1c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="color: rgb(114, 128, 157);"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.88 9.29L12 13.17 8.12 9.29a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41-.39-.38-1.03-.39-1.42 0z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</header>
`;
